src/k2/httpproxy/http_proxy.h

```diff
@@ -21,18 +21,21 @@ Copyright(c) 2021 Futurewei Cloud
     SOFTWARE.
```

Comment:
- You should name the files with UpperCamelCase.
- The header file should be split into header and implementation.  All implementation should move from the header to  the .cpp file.
- this folder needs a CMakeLists.txt file
- this folder needs to be added to the parent CMakeLists.txt file

Reply:
Pushed a new commit where I renamed `http_proxy.h` to `HTTPProxy.h` and moved implementation to `HTTPProxy.cpp` and created library `libhttproxy`. Please review again.